### PR TITLE
Update inferRelationDetails method to incorporate changes in one-one associations

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:[4163](https://github.com/ballerina-platform/ballerina-standard-library/issues/4163) [4228](https://github.com/ballerina-platform/ballerina-standard-library/issues/4228)

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 1]
```diff
@@ -254,10 +254,21 @@ public static void inferRelationDetails(Module entityModule) {
...
+                                                field.setRelation(computeRelation(entity, assocEntity, true,
+                                                        Relation.RelationType.ONE));
+                                                assocfield.setRelation(computeRelation(assocEntity, entity, false,
+                                                        Relation.RelationType.ONE));
+                                            } else if (field.isOptionalType() && !assocfield.isOptionalType()) {
+                                                field.setRelation(computeRelation(entity, assocEntity, false,
+                                                        Relation.RelationType.ONE));
+                                                assocfield.setRelation(computeRelation(assocEntity, entity, true,
+                                                        Relation.RelationType.ONE));
+                                            } else {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:267]: `if (!field.isOptionalType() && assocfield.isOptionalType())` is this condition covered here

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:267]: yes, it's covered in line 257 right?

**Review by daneshk — DISMISSED** [review]

**Review by daneshk — DISMISSED** [review]

File: persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java [hunk 2]
```diff
@@ -167,7 +158,17 @@ public void execute() {
             }
             updateBallerinaToml(moduleName, datastore);
         } catch (BalException e) {
-            errStream.println("ERROR: failed to add database configurations in the toml file. " + e.getMessage());
+            errStream.println("ERROR: failed to add persist configurations in the toml file. " + e.getMessage());
+            return;
+        }
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java:163]: Shall we move updating the toml file, before creating the persist directory? If persist config exists, we failed the operation but we create persist directory.

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java:135]: updated in [2ebbe32](https://github.com/ballerina-platform/persist-tools/pull/174/commits/2ebbe32a5b4761a44d3f598df2e04221697ddbeb)

**Pull request merged**